crypto/rand/rand_lib.c

```diff
@@ -42,134 +42,126 @@ int rand_fork_count;
  * it's not sufficient to indicate whether or not the seeding was
  * done.
  */
-void rand_read_tsc(RAND_poll_cb rand_add, void *arg)
+void rand_read_tsc(RAND_POOL * pool)
 {
     unsigned char c;
     int i;
 
     if ((OPENSSL_ia32cap_P[0] & (1 << 4)) != 0) {
         for (i = 0; i < TSC_READ_COUNT; i++) {
             c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
-            rand_add(arg, &c, 1, 0.5);
+            RAND_POOL_add(pool, &c, 1, 0.5);
```

Comment:
I think we should just add the whole TSC, I see little value in only taking the last 8 bits.

Reply:
I guess you can leave it like this for now.

Reply:
I prefer adding the whole of it.  There is some randomness in the bits outside of the last 8.  Not a lot but some.  My gathering xors the bytes together to produce a single byte of output but it isn't necessary.

Reply:
If you don't mind, I would prefer if you could postpone this question to a followup PR, because it's only indirectly related to the subject of this PR.